bin/demo.py

```diff
@@ -31,97 +33,75 @@
-            headers['Authorization'] = "token={}".format(token_arg)
-        return headers
-    return auth_wrapper
-
-auth_func = get_auth_wrapper(None)
-
-def check_and_set_token(jenkins_url):
+def check_and_set_token():
+    dcos_username = arguments['--dcos-username']
+    dcos_password = arguments['--dcos-password']
     try:
-        global auth_func
-        command = "dcos config show core.dcos_acs_token"
-        token = check_output(command, shell=True).strip('\n')
-        auth_func = get_auth_wrapper(token)
-        r = requests.get(jenkins_url, headers=auth_func({}), verify=False)
-        if r.status_code == 401:
-            log_and_exit("Not authenticated. Please run `dcos auth login` and try again.")
-    except CalledProcessError:
-        log_and_exit ("Not authenticated. Please run `dcos auth login` and try again.")
+        token = shakedown.authenticate(dcos_username, dcos_password)
+        dcos.config.set_val('core.dcos_acs_token', token)
+    except:
+        log_and_exit('!! authentication failed')
 
 def config_dcos_cli(dcos_url):
-    if call (["dcos", "config", "set", "core.dcos_url", dcos_url],stdout=open(os.devnull, 'wb')) == 1:
-        log_and_exit ("Unable to configure DCOS CLI.")
-
-def make_temp_dir():
-    remove_temp_dir()
-    os.mkdir("tmp")
-
-def rename(path, jenkins_name):
-    with open(path, 'r+') as f:
-        config = json.load(f)
-        config['jenkins']['framework-name'] = jenkins_name
-        f.seek(0)
-        json.dump(config, f, indent=4)
-
-def install(dcos_url, jenkins_name, jenkins_url):
-    log ("Installing Jenkins with name {}.".format(jenkins_name))
-    shutil.copyfile("conf/jenkins.json", "tmp/jenkins.json")
-    rename("tmp/jenkins.json", jenkins_name)
-    command = "dcos package install --yes --options=tmp/jenkins.json jenkins"
-    print ("\n> " + command)
-    if call (['dcos', 'package', 'install', '--yes', '--options=tmp/jenkins.json', 'jenkins']) != 0:
-        log_and_exit ("Failed to install Jenkins.")
-    log("Jenkins has been installed! Wait for it to come up before proceeding at: {}".format(jenkins_url))
-
-def verify(jenkins_url):
-    r = requests.get(jenkins_url, headers=auth_func({}), verify=False)
-    if r.status_code != 200:
-        log ("Couldn't find a Jenkins instance running at {}.".format(jenkins_url))
+    dcos.config.set_val('core.dcos_url', dcos_url)
+    dcos.config.set_val('core.ssl_verify', 'False')
+
+def install_jenkins(jenkins_name, jenkins_url):
+    log("installing Jenkins with name '{}'".format(jenkins_name))
+    install_package_and_wait('jenkins', None, jenkins_name)
+    assert package_installed('jenkins'), log_and_exit('!! package failed to install')
+    log("waiting for Jenkins service to come up at '{}'".format(jenkins_url))
+    end_time = time.time() + 300
+    while time.time() < end_time:
+        if verify_jenkins(jenkins_url):
+            break
+        time.sleep(1)
+
+def verify_jenkins(jenkins_url):
```

Comment:
Yeah, the [authenticated] `http` object being used (from the `dcos` package) throws an exception for anything but a `2xx` response -- so this will return false for a `503`.